File: ts/src/binarySearch.ts
const binarySearch = (list: number[], target: number) => {
	let low = 0
	let high = list.length - 1

	while (low <= high) {
		const mid = Math.ceil((high + low) / 2)
		const guess = list[mid]

		if (guess === target) return mid

		if (guess > target) high = mid - 1
		else low = mid + 1
	}
}

const array = Array.from({ length: 128 }, (_, i) => ++i)

console.log(binarySearch(array, 50))

File: py/binary_search.py
import math


def binary_serch(items, target):
    low = 0
    high = len(items) - 1

    while low <= high:
        mid = math.ceil((high + low) / 2)
        candidate = items[mid]

        if candidate == target:
            return mid

        if candidate > target:
            high = mid - 1
        else:
            low = mid + 1


print(binary_serch([1, 2, 3, 4, 5, 6, 7, 8], 6))
